Mathlib/Tactic/Linarith/Parsing.lean

```diff
@@ -94,9 +94,16 @@ def Sum.mul (s1 s2 : Sum) : Sum :=
     RBMap.empty
 
 /-- The `n`th power of `s : Sum` is the `n`-fold product of `s`, with `s.pow 0 = Sum.one`. -/
-def Sum.pow (s : Sum) : ℕ → Sum
-  | 0     => Sum.one
-  | (k+1) => s.mul (s.pow k)
+partial def Sum.pow (s : Sum) : ℕ → Sum
+  | 0 => Sum.one
+  | 1 => s
+  | n =>
+    let m := n >>> 1
+    let a := s.pow m
+    if n &&& 1 == 0 then
```

Comment:
Presumably this should either be
```suggestion
    if n &&& 1 = 0 then
```
or
```suggestion
    bif n &&& 1 == 0 then
```
?

Reply:
Okay, but for the record I think it is dumb that these variations exist